### PR TITLE
ci: migrate to GH actions

### PR DESCRIPTION
**Pull request by saihaj**
### Summary of PR
Migrate CCI workflow to GH actions. I split it into 2 workflows one for CI which runs tests and listing and CD for building and deploying to NPM

### For Maintainers
* [x] Setup secret `NPM_BOT_TOKEN`

### Time spent on PR
30 mins

### Demo
For some reason `npm run publish-coverage` is not working. I don't know if this is something I need add a secret or something for coveralls.
https://github.com/saihaj/ShabadOSgurmukhi-utils/actions/runs/324357425

### Linked issues
Ongoing migration process 

### Reviewers
@Harjot1Singh @bhajneet

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
+
+      - name: Install npm dependencies
+        run: npm ci
+
+      - name: Run Tests
+        run: npm run test
+
+      - name: Publish Coverage Report
```
Comment (Harjot1Singh) [inline .github/workflows/ci.yml:71]: Scrap this, remove it from package.json, use https://github.com/marketplace/actions/coveralls-github-action instead. Having this command is not useful offline.

**Review by Harjot1Singh — CHANGES_REQUESTED** [review]

**Comment by saihaj** [comment]
I rebased mine with dev. I think it probably has something to do with readme 🤷🏼‍♂️ ?

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
   },
   "scripts": {
     "test": "jest",
-    "publish-coverage": "jest --coverage && cat ./coverage/lcov.info | coveralls",
+    "test:coverage": "jest --coverage",
```
Comment (Harjot1Singh) [inline package.json:16]: tbh, i'd just scrap this and add coverage by default to the test command, and just use that

**Review by Harjot1Singh — CHANGES_REQUESTED** [review]
Great work, just this one change!

**Review by Harjot1Singh — APPROVED** [review]

**Pull request merged**